### PR TITLE
Support GHC 9.2

### PR DESCRIPTION
**Pull request by brandon-leapyear**

### CODE REVIEW

**Review by abhinav — APPROVED** [review]
Thanks! The changes appear pretty straightforward. 
CC @phile314

**Comment by brandon-leapyear** [comment]
FYI the fix for `pinch-gen` is here: https://github.com/phile314/pinch-gen/pull/7

File: tests/Pinch/ClientServerSpec.hs [hunk 1]
```diff
@@ -34,9 +35,9 @@ echoServer = createServer $ \_ -> Just $ CallHandler $ \_ (r :: Value TStruct) -
   pure r
 
 data CalcRequest = CalcRequest
-  { inp1 :: Field 1 Int32
-  , inp2 :: Field 2 Int32
-  , op   :: Field 3 Op
+  { _inp1 :: Field 1 Int32
+  , _inp2 :: Field 2 Int32
+  , _op   :: Field 3 Op
```
Comment (phile314) [inline tests/Pinch/ClientServerSpec.hs:40]: Hm, what's the point of this rename? Is this silencing some warnings?

File: tests/Pinch/TransportSpec.hs [hunk 2]
```diff
@@ -19,7 +19,7 @@ import qualified Pinch.Internal.Builder as B
 
 data MemoryConnection = MemoryConnection
   { contents :: IORef ByteString
-  , maxChunkSize :: Int -- ^ how many bytes to maximally return for one cGetSome call
+  , _maxChunkSize :: Int -- ^ how many bytes to maximally return for one cGetSome call
```
Comment (phile314) [inline tests/Pinch/TransportSpec.hs:22]: dito

**Review by phile314 — APPROVED** [review]
Looks good to me.

**Comment by phile314** [comment]
@brandon-leapyear Can you add GHC 9.2 to https://github.com/abhinav/pinch/blob/master/.github/workflows/build.yaml ?

File: tests/Pinch/ClientServerSpec.hs (hunk 1 above)
Comment (brandon-leapyear) [reply tests/Pinch/ClientServerSpec.hs:40]: Yeah; there's some name shadowing, e.g. https://github.com/abhinav/pinch/blob/2578dc3114b443e6f1a19ca22f715edb38972001/tests/Pinch/ClientServerSpec.hs#L55

Personally, I would use record wild cards or something, but this was the easiest change

File: tests/Pinch/TransportSpec.hs (hunk 2 above)
Comment (brandon-leapyear) [reply tests/Pinch/TransportSpec.hs:22]: `contents` is used in `mGetContents`, but `maxChunkSize` was never used, so it was throwing unused-top-bindings warnings. Again, solvable in other ways, this is the easiest solution

**Comment by brandon-leapyear** [comment]
https://github.com/abhinav/pinch/pull/45/files#diff-d0777657fa3fd81d23aaf7273e58aee453b04e67882517900c56daeef9b3e4c1R8

Does this not work?

**Comment by abhinav** [comment]
image

The linked changes were enough. Tests ran against GHC 9.2.1.

**Pull request merged**